extensions/mongodb/runtime/src/main/java/org/apache/camel/quarkus/component/mongodb/CamelMongoClientProducer.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.quarkus.component.mongodb;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.inject.Produces;
+import javax.inject.Named;
+import javax.inject.Singleton;
+
+import com.mongodb.client.MongoClient;
+
+@ApplicationScoped
+public class CamelMongoClientProducer {
+
+    @Produces
+    @Singleton
+    @Named("camelMongoClient")
+    public com.mongodb.MongoClient camelMongoClient(MongoClient delegate) {
```

Comment:
Can't we install the `com.mongodb.MongoClient` into the Camel registry using `MongoClientBuildItem` ? Avoiding CDI could bring some performance benefits.

Reply:
I'll see if it's possible.

Reply:
Latest changes remove the CDI bits in favour of `MongoClientBuildItem`.